backend/pkg/api/connect/integration/transform_test.go

```diff
@@ -0,0 +1,259 @@
+	"github.com/redpanda-data/console/backend/pkg/config"
+	v1alpha1 "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1"
+	v1alpha1connect "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1/dataplanev1alpha1connect"
+)
+
+var (
+	technicallyATransform = []byte{0x00, 0x61, 0x73, 0x6D}
+	transformTimeout      = 30 * time.Second
+)
+
+func despawnTopic(ctx context.Context, client v1alpha1connect.TopicServiceClient, topic string) error {
+	_, err := client.DeleteTopic(ctx, connect.NewRequest(&v1alpha1.DeleteTopicRequest{
+		Name: topic,
+	}))
+	return err
+}
+
+func spawnTopic(ctx context.Context, partitionCount int32, client v1alpha1connect.TopicServiceClient, topic string) error {
+	_, err := client.CreateTopic(ctx, connect.NewRequest(&v1alpha1.CreateTopicRequest{
+		Topic: &v1alpha1.CreateTopicRequest_Topic{
+			Name:           topic,
+			PartitionCount: &partitionCount,
+			Configs: []*v1alpha1.CreateTopicRequest_Topic_Config{
+				{
+					Name:  "cleanup.policy",
+					Value: kadm.StringPtr("compact"),
+				},
+			},
+		},
+	}))
+	return err
+}
+
+func spawnTransform(cfg config.Server, name, inputTopic string, outputTopics []string, env []adminapi.EnvironmentVariable, wasm []byte) error {
+	metadata := api.TransformMetadata{
+		Name:         name,
+		InputTopic:   inputTopic,
+		OutputTopics: outputTopics,
+		Environment:  env,
+	}
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+
+	metadataPart, err := writer.CreateFormField("metadata")
+	if err != nil {
+		return err
+	}
+
+	if err := json.NewEncoder(metadataPart).Encode(metadata); err != nil {
+		return err
+	}
+
+	wasmPart, err := writer.CreateFormFile("wasm_binary", "transform.wasm")
+	if err != nil {
+		return err
+	}
+
+	if _, err := wasmPart.Write(wasm); err != nil {
+		return err
+	}
+
+	if err := writer.Close(); err != nil {
+		return err
+	}
+
+	req, err := http.NewRequestWithContext(context.Background(), "PUT", fmt.Sprintf("http://%s:%d/v1alpha1/transforms", cfg.HTTPListenAddress, cfg.HTTPListenPort), body)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", writer.FormDataContentType())
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusCreated {
+		return fmt.Errorf("unexpected status code: %d message: %s", resp.StatusCode, resp.Status)
+	}
+	return nil
+}
+
+func (s *APISuite) TestDeployTransform() {
+	t := s.T()
+
+	require := rqr.New(t)
+	assert := asrt.New(t)
+
+	t.Run("create transform with valid request (connect-go)", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), transformTimeout)
+		defer cancel()
+
+		topicClient := v1alpha1connect.NewTopicServiceClient(http.DefaultClient, s.httpAddress())
+		require.NoError(spawnTopic(ctx, 2, topicClient, "wasm-tfm-create-test-i"))
+		require.NoError(spawnTopic(ctx, 3, topicClient, "wasm-tfm-create-test-o"))
```

Comment:
can we maybe change the tests to do deferred despawnTopic of created topics after we have successfully created the topics. so it's clearer that the deferred topic cleanup is immediately after creation in the code.